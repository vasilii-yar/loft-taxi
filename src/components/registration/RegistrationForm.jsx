import React, {useState} from 'react';
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import Input from "@material-ui/core/Input";
import Button from "@material-ui/core/Button";
import Paper from "@material-ui/core/Paper";
import "./RegistrationForm.css";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {tryRegistering} from "../../redux/modules/auth/authActions";
import {Link, Redirect, withRouter} from "react-router-dom";
import {useForm} from "react-hook-form";
import TextField from "@material-ui/core/TextField";
import {FieldsValidateErrorMessages} from "../fieldsvalidate/FieldsValidateErrorMessages";

const RegistrationForm = (props) => {
    const [state, setState] = useState({
        email: "",
        name: "",
        surname: "",
        password: ""
    });

    const {register, handleSubmit, errors} = useForm();

    const onSubmit = (data) => {
        props.tryRegistering(state.email, state.password, state.name, state.surname);
    }

    const handleChange = (event) => {
        const target = event.target;
        const name = target.name;
        const value = target.value;

        setState(
            {...state, [name]: value}
        );
    }

    return (
        props.isLoggedIn ?
            (
                <>
                    <Redirect to="/map"/>
                </>
            )
            :
            (
                <Paper>
                    <Container>
                        <Typography variant="h4" gutterBottom className="login-header">
                            Регистрация
                        </Typography>
                        <form onSubmit={handleSubmit(onSubmit)}>
                            <Grid container direction={"column"} spacing={2}>
                                <Grid item xs={12}>
                                    <Typography>
                                        Уже зарегистрированы? <Link to="/login">Войти</Link>
                                    </Typography>
                                </Grid>
                                <Grid item xs={12}>
                                    <TextField inputRef={register({required: true})}
                                               name="email"
                                               type="email"
                                               value={state.email}
                                               onChange={handleChange}
                                               fullWidth
                                               label="Адрес электронной почты"
                                               error={errors.email}
                                    />
                                    <FieldsValidateErrorMessages error={errors.email}/>
                                </Grid>
                                <Grid item xs={12}>
                                    <Grid container direction="row">
                                        <Grid item xs={5}>
                                            <TextField inputRef={register({required: true})}
                                                       name="name"
                                                       type="text"
                                                       value={state.name}
                                                       onChange={handleChange}
                                                       fullWidth
                                                       label="Имя"
                                                       error={errors.name}
                                            />
                                            <FieldsValidateErrorMessages error={errors.name}/>
                                        </Grid>
                                        <Grid item xs={2}></Grid>
                                        <Grid item xs={5}>
                                            <TextField inputRef={register({required: true})}
                                                       name="surname"
                                                       type="text"
                                                       value={state.surname}
                                                       onChange={handleChange}
                                                       fullWidth
                                                       label="Фамилия"
                                                       error={errors.surname}
                                            />
                                            <FieldsValidateErrorMessages error={errors.surname}/>
                                        </Grid>
                                    </Grid>
                                </Grid>
                                <Grid item xs={12}>
                                    <TextField inputRef={register({required: true})}
                                               name="password"
                                               type="password"
                                               value={state.password}
                                               onChange={handleChange}
                                               fullWidth
                                               label="Пароль"
                                               error={errors.password}
                                    />
                                    <FieldsValidateErrorMessages error={errors.password}/>
                                </Grid>
                                <Grid item xs={12}>
                                    <div className="login-button">
                                        <Button variant="contained" color="primary" type="submit">
                                            Войти
                                        </Button>
                                    </div>
                                </Grid>
                            </Grid>
                        </form>
                    </Container>
                </Paper>
            )
    );
}

RegistrationForm.propTypes = {
    tryRegistering: PropTypes.func.isRequired,
    isLoggedIn: PropTypes.bool.isRequired
}

const mapStateToProps = (state) => {
    return (
        {isLoggedIn: state.auth.isLoggedIn}
    )
}

export default connect(
    mapStateToProps,
    {tryRegistering}
)(withRouter(RegistrationForm));