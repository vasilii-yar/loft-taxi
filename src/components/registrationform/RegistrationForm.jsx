import React from 'react';
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import Input from "@material-ui/core/Input";
import Button from "@material-ui/core/Button";
import Paper from "@material-ui/core/Paper";
import "./RegistrationForm.css";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {tryRegistering} from "../../actions/authActions";
import {Link, withRouter} from "react-router-dom";

class RegistrationForm extends React.Component {
    static propTypes = {
        logIn: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props);
        this.state = {
            email: "",
            name: "",
            surname: "",
            password: ""
        }
    }

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.tryRegistering(this.state.email, this.state.password, this.state.name, this.state.surname);
    }

    goToMap = () => {
        this.props.history.push("/map");
    }

    handleChange = (event) => {
        const target = event.target;
        const name = target.name;
        const value = target.value;

        this.setState(
            {[name]: value}
        )
    }

    render() {
        return (
            this.props.isLoggedIn ?
                (
                    <>
                        {this.goToMap()}
                    </>
                )
                :
                (
                    <Paper>
                        <Container>
                            <Typography variant="h4" gutterBottom className="login-header">
                                Регистрация
                            </Typography>
                            <form onSubmit={this.handleSubmit}>
                                <Grid container direction={"column"} spacing={2}>
                                    <Grid item xs={12}>
                                        <Typography>
                                            Уже зарегистрированы? <Link to="/login">Войти</Link>
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormControl fullWidth>
                                            <InputLabel htmlFor="login">Адрес электронной почты</InputLabel>
                                            <Input
                                                id="email"
                                                name="email"
                                                type="email"
                                                onChange={this.handleChange}
                                                fullWidth
                                            />
                                        </FormControl>
                                    </Grid>
                                    <Grid item xs={12}>
                                        <Grid container direction="row">
                                            <Grid item xs={5}>
                                                <FormControl fullWidth>
                                                    <InputLabel htmlFor="login">Имя</InputLabel>
                                                    <Input
                                                        id="name"
                                                        name="name"
                                                        type="text"
                                                        onChange={this.handleChange}
                                                    />
                                                </FormControl>
                                            </Grid>
                                            <Grid item xs={2}></Grid>
                                            <Grid item xs={5}>
                                                <FormControl fullWidth>
                                                    <InputLabel htmlFor="login">Фамилия</InputLabel>
                                                    <Input
                                                        id="surname"
                                                        name="surname"
                                                        type="text"
                                                        onChange={this.handleChange}
                                                    />
                                                </FormControl>
                                            </Grid>
                                        </Grid>
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormControl fullWidth>
                                            <InputLabel htmlFor="password">Пароль</InputLabel>
                                            <Input
                                                id="password"
                                                name="password"
                                                type="password"
                                                onChange={this.handleChange}
                                                fullWidth
                                            />
                                        </FormControl>
                                    </Grid>
                                    <Grid item xs={12}>
                                        <div className="login-button">
                                            <Button variant="contained" color="primary" type="submit">
                                                Войти
                                            </Button>
                                        </div>
                                    </Grid>
                                </Grid>
                            </form>
                        </Container>
                    </Paper>
                )
        );
    }
}

export default connect(
    (state) => ({isLoggedIn: state.auth.isLoggedIn}),
    {tryRegistering}
)(withRouter(RegistrationForm));