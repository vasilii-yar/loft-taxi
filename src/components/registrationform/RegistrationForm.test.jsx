import React from 'react';
import {render} from "@testing-library/react";
import RegistrationForm from "./RegistrationForm";

describe("Profile", () => {
    it("renders correctly", () => {
        const {container} = render(<RegistrationForm/>);
        expect(container.innerHTML).toMatch("Регистрация");
    })
});