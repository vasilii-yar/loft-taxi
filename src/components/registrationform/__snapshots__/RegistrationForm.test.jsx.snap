// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly check by snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h4
      className="MuiTypography-root login-header MuiTypography-h4 MuiTypography-gutterBottom"
    >
      Регистрация
    </h4>
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Уже зарегистрированы? 
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Войти
            </a>
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="login"
            >
              Адрес электронной почты
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
            >
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  htmlFor="login"
                >
                  Имя
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiInput-input"
                    disabled={false}
                    id="name"
                    name="name"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            />
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
            >
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  htmlFor="login"
                >
                  Фамилия
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiInput-input"
                    disabled={false}
                    id="surname"
                    name="surname"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="password"
            >
              Пароль
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="login-button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Войти
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
